controller/controller.go

```diff
@@ -136,19 +136,32 @@ func (c *controller) Start() error {
 		startedUpdaters = append(startedUpdaters, u)
 	}
 
-	c.watchForUpdates()
+	err := c.watchForUpdates()
+	if err != nil {
+		return err
+	}
 
 	c.started = true
 	return nil
 }
 
-func (c *controller) watchForUpdates() {
-	ingressWatcher := c.client.WatchIngresses()
-	serviceWatcher := c.client.WatchServices()
-	namespaceWatcher := c.client.WatchNamespaces()
+func (c *controller) watchForUpdates() error {
+	ingressWatcher, err := c.client.WatchIngresses()
+	if err != nil {
+		return err
+	}
+	serviceWatcher, err := c.client.WatchServices()
+	if err != nil {
+		return err
+	}
+	namespaceWatcher, err := c.client.WatchNamespaces()
+	if err != nil {
+		return err
+	}
 	c.watcher = k8s.CombineWatchers(ingressWatcher, serviceWatcher, namespaceWatcher)
 	c.watcherDone.Add(1)
 	go c.handleUpdates()
+	return nil
 }
```

Comment:
This is actually changing Feed's startup behaviour. 
Before, Feed would start up, even if there were sync errors on the k8s.client resources watchers. If the client remained `unsynced`, this would be caught on the `updateIngresses` cycle, and set feed in an unhealthy state.
 
Now, if for some reason Feed gets an error while starting to watch the resources, Feed will error and not start.
This could send feed into a crash loop state depending on the cause of the error. 

Although undocumented and untested, I don't think we should change that behaviour.

Reply:
I agree. The client api was actually designed to make it impossible for Watcher(s) to return errors, so feed doesn't have to rely on the apiserver to be up and running to start. In fact it is expected that errors are handled during the control loop, so get calls that fail due to cache not being fully synced or other can be retried.